docs/persistency.md

```diff
@@ -160,3 +160,117 @@ This is equivalent to the following command executed inside the container:
+                batch_op.drop_constraint('Task_simulation_id_fkey', type_='foreignkey')
+                batch_op.create_foreign_key(None, 'Simulation', ['simulation_id'], ['id'], ondelete='CASCADE')
+        ```
+        In this case change None to 'Task_simulation_id_fkey'.
+8. Run `flask --app yaptide.application db upgrade` to apply migration script.
+9. To undo changes and go back to previous version run `flask --app yaptide.application db downgrade`.
+10. Commit and push script and modification to models.py.
+
+### Testing migration script with copy of production volume
+
+1. If there is testing environment other than local - pull changes.
+2. Copy volume ("data") of production postgres database and save it under different name.
+3. In `docker-compose.yml` modify database configuration to use this volume.
+4. Run `scripts/start_with_docker_develop.sh`, to run backend and additionaly pgadminer tool (see section Using pgadminer).
+5. Again prepare `FLASK_SQLALCHEMY_DATABASE_URI` like above and use it together with each `flask db` command.
+6. Run `flask --app yaptide.application db upgrade`.
+7. Check also `flask --app yaptide.application db downgrade`, then do upgrade again. All should execute without errors.
+8. Do manual testing. Check functionalities, some might be unnecessary to check depending which part in `models.py` was changed:
+    - logging in and out
+    - Loading simulation results, input files, logs.
+    - Submiting new simulation
+    - operations contained in `admin/db_manage.py`
+
+### Migrating production
+1. Run git pull on master.
+2. Backup the production database
+  Before applying any migrations, create a backup of the live database:
+    `
+    pg_dump -U <db_user> -h <db_host> -d <db_name> > backup.sql
+    `
+3. Again prepare `FLASK_SQLALCHEMY_DATABASE_URI` like above and use it together with each `flask db` command.
+4. Applying the migration in production
+    There are two options for applying the migration:
+
+    - Option 1: Execute from outside the Docker container
+
+      `
+      FLASK_SQLALCHEMY_DATABASE_URI=postgresql+psycopg://user:password@<db_container_ip>:5432/db_name flask --app yaptide.application db upgrade
+      `
+    - Option 2: Execute from inside the Flask container
+
+    Access the container and run the upgrade:
+      `
+      docker exec -it <flask_container> bash
+      `
+      `
+      FLASK_SQLALCHEMY_DATABASE_URI=postgresql+psycopg://user:password@<db_container_ip>:5432/db_name flask --app yaptide.application db upgrade
+      `
+    Rollback strategy
+    In case of any issues, you can revert the changes by running:
+      `
+      flask --app yaptide.application db downgrade
+      `
+    Post-migration testing
+    Perform manual tests on the production system:
+
+    - Verify logging in and out.
+    - Check simulation submissions.
+    - Ensure any functionality affected by the migration is working.
+
+
+5. In case of restoring database from backup, run:
```

Comment:
This can be done as well as restoring the volume itself:

```
docker run --rm -v yaptide_data:/var/lib/postgresql/data -v /home/ubuntu/backup:/backup busybox tar xzf /backup/yaptide_data_backup.tar.gz -C /var/lib/postgresql/data
```